packages/react/src/components/DataTable/__tests__/TableSelectRow-test.js

```diff
@@ -23,45 +23,49 @@ describe('DataTable.TableSelectRow', () => {
     };
   });
 
-  it('should render', () => {
-    const wrapper = mount(
-      <Table>
-        <TableHead>
-          <TableRow>
-            <TableSelectRow {...mockProps} />
-          </TableRow>
-        </TableHead>
-      </Table>
-    );
-    expect(wrapper).toMatchSnapshot();
-  });
+  describe('renders as expected - Component API', () => {
+    it('should render', () => {
+      const { container } = render(
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableSelectRow {...mockProps} />
+            </TableRow>
+          </TableHead>
+        </Table>
+      );
+      expect(container).toMatchSnapshot();
+    });
 
-  it('should render with the provided class name', () => {
-    const customClassName = 'custom-table-select-row-classname';
-    const wrapper = mount(
-      <Table>
-        <TableHead>
-          <TableRow>
-            <TableSelectRow {...mockProps} className={customClassName} />
-          </TableRow>
-        </TableHead>
-      </Table>
-    );
-    const elements = wrapper.find(`td.${customClassName}`);
-    expect(elements.length).toBe(1);
+    it('should render with the provided class name', () => {
+      const customClassName = 'custom-table-select-row-classname';
+      const { container } = render(
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableSelectRow {...mockProps} className={customClassName} />
+            </TableRow>
+          </TableHead>
+        </Table>
+      );
+      const elements = container.querySelectorAll(`td.${customClassName}`);
+      expect(elements.length).toBe(1);
+    });
   });
 
-  it('should invoke `onSelect` when clicked', () => {
-    const wrapper = mount(
-      <Table>
-        <TableHead>
-          <TableRow>
-            <TableSelectRow {...mockProps} />
-          </TableRow>
-        </TableHead>
-      </Table>
-    );
-    wrapper.find('input').simulate('click');
-    expect(mockProps.onSelect).toHaveBeenCalledTimes(1);
+  describe('behaves as expected', () => {
+    it('should invoke `onSelect` when clicked', () => {
+      const { container } = render(
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableSelectRow {...mockProps} />
+            </TableRow>
+          </TableHead>
+        </Table>
+      );
+      container.querySelector('input.cds--checkbox').click();
```

Comment:
Same thing here with using a query instead of querySelector. I think `screen.getByRole('checkbox')` would give you the same element to click. Also events should be fired using `user-event`.

`userEvent.click(screen.getByRole('checkbox'));`

Reply:
done!